db/database.go

```diff
@@ -6,65 +6,99 @@ import (
 	"fmt"
 	"github.com/exaring/otelpgx"
 	"github.com/jackc/pgx/v5"
+	"github.com/jackc/pgx/v5/pgconn"
 	"github.com/jackc/pgx/v5/pgxpool"
 	"github.com/wwi21seb-projekt/alpha-shared/config"
-
-	log "github.com/sirupsen/logrus"
+	"go.uber.org/zap"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"os"
 )
 
 type DB struct {
-	Pool *pgxpool.Pool
+	pool   *pgxpool.Pool
+	logger *zap.SugaredLogger
 }
 
-type TxFunc func(tx pgx.Tx) error
-
-func NewDB(ctx context.Context, dbCfg config.DatabaseConfig) (*DB, error) {
+func NewDB(ctx context.Context, dbCfg config.DatabaseConfig, logger *zap.SugaredLogger) (*DB, error) {
 	dsn := fmt.Sprintf("host=%s port=%s dbname=%s user=%s password=%s sslmode=disable search_path=%s",
 		dbCfg.PostgresHost, dbCfg.PostgresPort, dbCfg.PostgresDB, dbCfg.PostgresUser, dbCfg.PostgresPassword, dbCfg.SchemaName)
 
 	cfg, err := pgxpool.ParseConfig(dsn)
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("failed to parse database config: %w", err)
 	}
 
-	cfg.ConnConfig.Tracer = otelpgx.NewTracer(otelpgx.WithTrimSQLInSpanName())
+	cfg.MaxConns = 20
+
+	options := []otelpgx.Option{otelpgx.WithIncludeQueryParameters(), otelpgx.WithTrimSQLInSpanName()}
```

Comment:
Können wir sagen, dass wir `WithIncludeQueryParameters` nur in `local` appenden?